source/plugins/devices/mp/mpy.js

```diff
@@ -80,20 +80,22 @@ def listdir(directory):
 	print(json.dumps(r))
 
 listdir ('${escape(folder)}')`;
+
 		let ls = null;
 		try {
 			let res = await this.execute (cmd);
 			if (!res.stderr) {
 				ls = JSON.parse (res.stdout);
 			}
-			// else
-			// {
-			// 	// TODO show notification
-			// }
+			else
+			{
+				// TODO show notification
+			}
 		}
 		catch (e)
 		{
 			// TODO show notification
```

Comment:
I suggest emitting an event using emit ('error', 'listdir', 'error name') so that the UI can show a notification.